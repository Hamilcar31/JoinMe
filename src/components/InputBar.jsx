import React from 'react';
import {
  View, StyleSheet, TextInput,
} from 'react-native';

export default function InputBar(props) {
  const { placeholder = 'placeholder', style = {}, textStyle = {}, ...otherProps } = props;

  // const [value, onChangeText] = React.useState("");

  return (
    <View style={styles.container}>
        <TextInput
          style={styles.input}
          // onChangeText={(text) => onChangeText(text)}
          // value={value}
          placeholder={placeholder}
          {...otherProps}
        />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 40,
    width: 300,
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',

    backgroundColor: 'white',

    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.32,
    shadowRadius: 5.46,

    elevation: 9,
  },
  input: {
    width: "90%",
    height: "100%",
  },
});
