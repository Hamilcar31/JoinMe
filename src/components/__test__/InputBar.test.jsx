import React from 'react';
import renderer from 'react-test-renderer';

import InputBar from '../InputBar.jsx';

describe('input bar components', () => {
  it('has placeholder', () => {
    // given
    const inputBar = <InputBar placeholder="title"/>;

    // when
    const tree = renderer.create(inputBar).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});