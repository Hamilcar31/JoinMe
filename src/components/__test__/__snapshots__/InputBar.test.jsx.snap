// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input bar components has placeholder 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "white",
      "borderRadius": 25,
      "borderWidth": 1,
      "elevation": 9,
      "height": 40,
      "justifyContent": "center",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.32,
      "shadowRadius": 5.46,
      "width": 300,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="title"
    rejectResponderTermination={true}
    style={
      Object {
        "height": "100%",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
